File: src/core/gpu.rs
const GRAPHICS_SIZE: usize = 64 * 32;

pub struct GPU {
    pub memory: [bool; GRAPHICS_SIZE],
    pub draw: bool
}

impl GPU {
    pub fn new() -> Self {
        Self {
            memory: [false; GRAPHICS_SIZE],
            draw: true
        }
    }
}

File: src/RetroC8/Sources/RetroC8/GPU.swift
//
//  File.swift
//  
//
//  Created by Glenn Hevey on 20/11/2023.
//

import Foundation

struct GPU {
    let graphicsSize: UInt = 64 * 32
    
    var memory: [Bool]
    var draw: Bool
    
    init() {
        memory = [Bool](repeating: false, count: Int(graphicsSize))
        draw = true //TODO: Confirm whether this can be false once working.
    }
    
    mutating func update(draw: Bool) {
        self.draw = draw
    }
    
}
